native/src/modules/app/components/App.js

```diff
@@ -15,11 +15,31 @@ import NavigatorContainer from '../containers/NavigatorContainer'
 import { SafeAreaProvider } from 'react-native-safe-area-context'
 import StaticServerProvider from '../../static-server/containers/StaticServerProvider'
 import { NavigationContainer } from '@react-navigation/native'
+import PermissionSnackbarContainer from '../../layout/containers/PermissionSnackbarContainer'
 
 const dataContainer: DataContainer = new DefaultDataContainer()
 const store: Store<StateType, StoreActionType> = createReduxStore(dataContainer)
 
 const App = () => {
+  const [routeName, setRouteName] = useState<?string>(null)
+  const [routeKey, setRouteKey] = useState<?string>(null)
+  const [cityCode, setCityCode] = useState<?string>(null)
+  const [languageCode, setLanguageCode] = useState<?string>(null)
```

Comment:
Shouldn' these two be in the redux state instead of the component state here? Isn't there duplicated data now?

Reply:
Yes, they are. But then we need the whole mapping similar to DashboardContainer and need to check every route (Category, Event, ....). It is already an ugly hack, so imo this does not make it worse.